lib/elixir/lib/module/checker.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Module.Checker do
+  def verify(module_map, _binary) do
+    state = :maps.with([:module, :file, :compile_opts], module_map)
```

Comment:
We should explicitly define the state with all fields we will use and then only update those fields, never add new ones. This will make it easier for us to know which fields are available at any given moment and be cheaper too. Something like:


```
%{
  module: module,
  file: file,
  compile_opts: compile_opts
} = module_map

state = %{
  module: module,
  file: file,
  compile_opts: compile_opts,
  function: nil
}
```

Note I have removed the `line` key because I don't think we will actually use it.